File: server/src/models/Prompt.ts
// Step 1: Declare the Prompt class which extends global.DbObject
// Step 2: Define the properties: id, created, updated, authenticatedUser, prompt, style, title

import { DbObject } from "../lib/db";

export class Prompt extends DbObject {
    promptCategory: number;
    authenticatedUser: number;
    prompt: string;
    style: string;
    title: string;

    audio: boolean;
}

File: app/lib/models/Prompt.dart

import 'Serializable.dart';

@reflector
class Prompt extends Serializable {
  int? id;
  int? created;
  int? updated;
  int? createdBy;
  int? updatedBy;
  int? promptCategory;
  String text = "";
  String title = "";
  String style = "";
  int? authenticatedUser;
  int? likesCount;
  String? uploadUrl;
  int? uploaded;
  int? sharedImage;
}
